clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryClient.java

```
@@ -0,0 +1,96 @@
+  private final DataRecoveryExecutor executor;
+
+  public DataRecoveryClient() {
+    this(new DataRecoveryExecutor());
+  }
+
+  public DataRecoveryClient(DataRecoveryExecutor module) {
+    this.executor = module;
+  }
+
+  public DataRecoveryExecutor getExecutor() {
+    return executor;
+  }
+
+  public void execute(DataRecoveryParams drParams, StoreRepushCommand.Params cmdParams) {
+    Set<String> storeNames = drParams.getRecoveryStores();
+    if (storeNames == null || storeNames.isEmpty()) {
+      LOGGER.warn("store list is empty, exit.");
+      return;
+    }
+    if (!confirmStores(storeNames)) {
+      return;
+    }
+
+    getExecutor().perform(storeNames, cmdParams);
+  }
+
+  public boolean confirmStores(Set<String> storeNames) {
+    LOGGER.info("stores to recover: " + storeNames);
+    LOGGER.info("Recover " + storeNames.size() + " stores, please confirm (yes/no) [y/n]:");
+    Scanner in = new Scanner(System.in);
+    String line = in.nextLine();
+    return line.equalsIgnoreCase("yes") || line.equalsIgnoreCase("y");
+  }
+
+  public static class DataRecoveryParams {
+    private final String controllerUrl;
+    private final String multiStores;
+    private final String recoveryCluster;
+    private final Set<String> recoveryStores;
+
+    public DataRecoveryParams(String controllerUrl, String multiStores, String recoveryCluster) {
+      this.controllerUrl = controllerUrl;
+      this.multiStores = multiStores;
+      this.recoveryCluster = recoveryCluster;
+      this.recoveryStores = calculateRecoveryStoreNames(this.controllerUrl, this.multiStores, this.recoveryCluster);
+    }
+
+    public Set<String> getRecoveryStores() {
+      return recoveryStores;
+    }
+
+    private Set<String> calculateRecoveryStoreNames(String controllerUrl, String multiStores, String recoveryCluster) {
+      Set<String> storeNames = null;
+      // Give high priority to multiStores, if it contains meaningful data, ignore recoveryCluster.
```

Can we fail fast if the operator specifies both params?
Let us also make it clear that the operator should only specify one of the params.

---

Or maybe we should just remove the whole cluster recovery at this moment since there could be many gotchas to blindly recover a whole cluster.

---

All right, I can remove the whole cluster recovery support for now.